test/validation/api/traffic_mngr/traffic_mngr.c

```diff
@@ -1679,6 +1685,17 @@ set_reqs_based_on_capas(odp_tm_requirements_t *req)
 
 	if (tm_capabilities.tm_queue_shaper_supported)
 		req->tm_queue_shaper_needed = true;
+
+	/* Since packet priority mode effects only scheduler tests
+	 * and in our scheduler tests, we only test with set queues under
+	 * same parent node, we are fine with any packet priority mode.
+	 * This is because we are concerned with packet priority obeyed
+	 * at immediate parent node only.
```

Comment:
I do not understand the explanation in the comment. The priority mode does not matter if the test is testing a scheduler in TM nodes that are directly connected to TM queues and not in TM nodes fed by other TM nodes that could change the priority on the way.

However, it looks to me that the test is testing scheduler in a TM node fed by other TM nodes. "node_1_3" is the one being tested, the node has child nodes which in turn have queues. The tests use one queue of each child node of node_1_3. If priority mode is overwrite, then the child nodes overwrite packet priority before packets reach the node being tested.

Reply:
In traffic_mngr_test_scheduler(), only test that is enabled is test_sched_queue_priority() with node "node_1_1_3".
So the test case is currently testing priority b/w child nodes of same parent node.  When the test cases are enabled, we can split it to not run on overwrite mode ?

Reply:
I missed the early return statement. It is correct that the potentially problematic scheduler tests are not run.

I think the comment is still a bit unclear. How about this:
```
        /* We can use any packet priority mode since it does not affect
         * our tests. Our scheduler test tests scheduling only in a node
         * directly connected to TM queues and such nodes see the original
         * packet priority before it could have been overwritten by any node.
         */
```